diff --git a/arch/powerpc/platforms/40x/ppc40x_simple.c b/arch/powerpc/platforms/40x/ppc40x_simple.c
index 2a050007bbae..f7b5decea9bf 100644
--- a/arch/powerpc/platforms/40x/ppc40x_simple.c
+++ b/arch/powerpc/platforms/40x/ppc40x_simple.c
@@ -20,6 +20,10 @@
 #include <asm/time.h>
 #include <asm/udbg.h>
 #include <asm/uic.h>
+#if defined(CONFIG_OBS600)
+#include <asm/dcr.h>
+#include <asm/dcr-regs.h>
+#endif
 
 #include <linux/init.h>
 #include <linux/of_platform.h>
@@ -64,7 +68,11 @@ static const char * const board[] __initconst = {
 static int __init ppc40x_probe(void)
 {
 	if (of_device_compatible_match(of_root, board)) {
+#if defined(CONFIG_OBS600)
+		mtdcri(SDR0, SDR0_MFR, mfdcri(SDR0, SDR0_MFR) & ~0x0c000000);
+#else
 		pci_set_flags(PCI_REASSIGN_ALL_RSRC);
+#endif
 		return 1;
 	}
 
diff --git a/drivers/net/ethernet/ibm/emac/phy.c b/drivers/net/ethernet/ibm/emac/phy.c
index aa070c063e48..2c408a0e13d3 100644
--- a/drivers/net/ethernet/ibm/emac/phy.c
+++ b/drivers/net/ethernet/ibm/emac/phy.c
@@ -72,6 +72,11 @@ int emac_mii_reset_phy(struct mii_phy *phy)
 	if ((val & BMCR_ISOLATE) && limit > 0)
 		phy_write(phy, MII_BMCR, val & ~BMCR_ISOLATE);
 
+#ifdef CONFIG_OBS600
+	phy_write(phy, 0x1c, 0xa410);
+	phy_write(phy, 0x1c, 0x8804);
+#endif
+
 	return limit <= 0;
 }
 
