Index: linux-5.4.101/arch/powerpc/boot/dts/obs600.dts
===================================================================
--- linux-5.4.101.orig/arch/powerpc/boot/dts/obs600.dts
+++ linux-5.4.101/arch/powerpc/boot/dts/obs600.dts
@@ -15,6 +15,7 @@
 /dts-v1/;
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
 
 / {
 	#address-cells = <1>;
@@ -163,33 +164,41 @@
 					reg = <0x00000000 0x00000000 0x08000000>;
 					#address-cells = <1>;
 					#size-cells = <1>;
-					partition@0 {
-						label = "kernel + initrd";
-						reg = <0x00000000 0x03de0000>;
-					};
-					partition@3de0000 {
-						label = "user config area";
-						reg = <0x03de0000 0x00080000>;
-					};
-					partition@3e60000 {
-						label = "user program area";
-						reg = <0x03e60000 0x04000000>;
-					};
-					partition@7e60000 {
-						label = "flat device tree";
-						reg = <0x07e60000 0x00080000>;
-					};
-					partition@7ee0000 {
-						label = "test program";
-						reg = <0x07ee0000 0x00080000>;
-					};
-					partition@7f60000 {
-						label = "u-boot env";
-						reg = <0x07f60000 0x00040000>;
-					};
-					partition@7fa0000 {
-						label = "u-boot";
-						reg = <0x07fa0000 0x00060000>;
+					partitions {
+						compatible = "fixed-partitions";
+						#address-cells = <1>;
+						#size-cells = <1>;
+
+						partition@0 {
+							label = "kernel + initrd";
+							reg = <0x00000000 0x03de0000>;
+							compatible = "openwrt,uimage", "denx,uimage";
+							openwrt,ih-type = <IH_TYPE_MULTI>;
+						};
+						partition@3de0000 {
+							label = "user config area";
+							reg = <0x03de0000 0x00080000>;
+						};
+						partition@3e60000 {
+							label = "user program area";
+							reg = <0x03e60000 0x04000000>;
+						};
+						partition@7e60000 {
+							label = "flat device tree";
+							reg = <0x07e60000 0x00080000>;
+						};
+						partition@7ee0000 {
+							label = "test program";
+							reg = <0x07ee0000 0x00080000>;
+						};
+						partition@7f60000 {
+							label = "u-boot env";
+							reg = <0x07f60000 0x00040000>;
+						};
+						partition@7fa0000 {
+							label = "u-boot";
+							reg = <0x07fa0000 0x00060000>;
+						};
 					};
 				};
 			};
